### ISSUE TITLE
How to show all children if a parent matched.

### ISSUE THREAD

**Issue by ansoncyx15** [brimdata/react-arborist#177]
Hi, currently if the children is matched, all the parents will also matched.
On the other hand, is there any way to show all the children if the parents is matched?

**Comment by jameskerr** [comment]
Yes, you can write your own search match function as shown here in the docs: https://github.com/brimdata/react-arborist#tree-filtering.

**Comment by ansoncyx15** [comment]
Hi, I had figured out to show all child element when the parent is matched the search term.

`searchMatch={(node, term) => {
            const nameExisted = (asset) => {
              if (asset.data.name.toLowerCase().includes(term.toLowerCase())) {
                return true;
              } else if (asset.parent.data.name) {
                return nameExisted(asset.parent);
              } else {
                return false;
              }
            };

            return nameExisted(node);
          }}`

**Comment by ansoncyx15** [comment]
HI I had figured out how to list all the child when parent name is matched.

Please consider code as below:

`searchMatch={(node,` searchTerm) => {
  const nameExisted = (asset) => {
   if (asset.data.name.toLowerCase().includes(term.toLowerCase())) {
    return true;
  } else if (asset.parent.data.name) {
    return nameExisted(asset.parent);
  } else {
  return false;
  }
  };
  
  return nameExisted(node);
}`

**Comment by TrevorBurnham** [comment]
Consolidating into #349, which tracks this. Closing here to keep the discussion in one place — please follow / weigh in on #349.
